mobile/src/main/java/org/openhab/habdroid/background/AlarmChangedReceiver.java

```diff
@@ -0,0 +1,72 @@
+import androidx.work.Constraints;
+import androidx.work.NetworkType;
+import androidx.work.OneTimeWorkRequest;
+import androidx.work.WorkManager;
+
+import org.openhab.habdroid.R;
+
+import static android.content.Context.NOTIFICATION_SERVICE;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_ID_SEND_ALARM_CLOCK;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_TAG_BACKGROUND;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_TAG_BACKGROUND_ERROR;
+import static org.openhab.habdroid.background.BackgroundUtils.WORKER_TAG_SEND_ALARM_CLOCK;
+import static org.openhab.habdroid.background.BackgroundUtils.createNotificationChannels;
+import static org.openhab.habdroid.background.BackgroundUtils.makeBackgroundNotification;
+import static org.openhab.habdroid.util.Constants.PREFERENCE_ALARM_CLOCK_ENABLED;
+
+public class AlarmChangedReceiver extends BroadcastReceiver {
+    private static final String TAG = AlarmChangedReceiver.class.getSimpleName();
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        Log.d(TAG, "onReceive() with intent " + intent.getAction());
+        if (!intent.getAction().equals("android.app.action.NEXT_ALARM_CLOCK_CHANGED")
+                || Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP
+                || ! PreferenceManager.getDefaultSharedPreferences(context)
+                .getBoolean(PREFERENCE_ALARM_CLOCK_ENABLED, true)) {
+            Log.d(TAG, "Don't send data");
+            return;
+        }
+
+        startAlarmChangedWorker(context);
+    }
+
+    public static void startAlarmChangedWorker(Context context) {
+        Log.d(TAG, "startAlarmChangedWorker()");
+        final Constraints constraints = new Constraints.Builder()
+                .setRequiredNetworkType(NetworkType.CONNECTED)
+                .build();
+        final OneTimeWorkRequest sendAlarmClockWorker =
+                new OneTimeWorkRequest.Builder(AlarmChangedWorker.class)
+                .setConstraints(constraints)
+                .addTag(WORKER_TAG_SEND_ALARM_CLOCK)
+                .build();
+
+        WorkManager workManager = WorkManager.getInstance();
+        NotificationManager nm =
+                (NotificationManager)context.getSystemService(NOTIFICATION_SERVICE);
+
+        Log.d(TAG, "Cancel previous worker");
+        workManager.cancelAllWorkByTag(WORKER_TAG_SEND_ALARM_CLOCK);
+        nm.cancel(NOTIFICATION_TAG_BACKGROUND_ERROR, NOTIFICATION_ID_SEND_ALARM_CLOCK);
+
+        Log.d(TAG, "Schedule worker");
+        workManager.enqueue(sendAlarmClockWorker);
+
+        createNotificationChannels(context);
+        Notification notification = makeBackgroundNotification(context,
+                R.string.waiting_for_network_to_send_alarm_clock, R.drawable.ic_alarm_grey_24dp,
+                false, null);
+        nm.notify(NOTIFICATION_TAG_BACKGROUND, NOTIFICATION_ID_SEND_ALARM_CLOCK, notification);
```

Comment:
Is that notification really necessary? I think notifying on failure (for retry) is OK, even though I'd probably not even do that (I'd just continue retrying), but notifying during the upload operation seems a bit much to me.

Reply:
When I added this I thought at the following scenario: A user accidentally disables network, sets up alarm and will never be notified, because Android doesn't run the worker at all.

Reply:
OK, makes sense.